feat(highlight): tweak default color scheme (#26540)
neovim/neovim: runtime/doc/news.txt

  • Legacy and extmark signs are displayed and listed with the same priority:
      line number -> priority -> sign id -> recently placed

• Default color scheme has been updated to be "Nvim branded" and accessible.
  Use `:colorscheme vim` to revert to the old legacy color scheme.

  Here is a list of Nvim specific highlight groups which are now defined in
  a meaningfully different way and might need an update:
  • |hl-FloatBorder| is linked to |hl-NormalFloat| instead of |hl-WinSeparator|.
  • |hl-NormalFloat| is not linked to |hl-Pmenu|.
  • |hl-WinBar| is linked to |hl-StatusLine|.
  • |hl-WinBarNC| is linked to |hl-StatusLineNC| instead of |hl-WinBar|.
  • |hl-WinSeparator| is linked to |hl-Normal| instead of |hl-VertSplit|.

>   This also might result into some color schemes looking differently.
>   It is due them relying on an implicit assumptions about how highlight groups

Reviewer: ```suggestion
  This also might result into some color schemes looking differently due to them
  relying on an implicit assumptions about how highlight groups
```

>   are defined by default. To overcome this, define all attributes of highlight

Reviewer: ```suggestion
  are defined by default. To account for this, define all attributes of highlight
```
